### PR TITLE
Fixes Wiki Book Manuals not working

### LINKED ISSUE #79545: Books that link to the wiki are broken
Reporting client version: 514.1589

## Round ID:
[218223](https://scrubby.melonmesa.com/round/218223)

## Testmerges:
- [Revert "Bumps rust-g to `3.0.0` "](https://www.github.com/tgstation/tgstation/pull/77307)
- [Extends Halloween](https://www.github.com/tgstation/tgstation/pull/79316)

## Reproduction:

Pick up space law book, try to read the book. Get HTML code instead of legible book, try with other books to similar results. 
```html
        <html>
        <head>
        <meta http-equiv='Content-Type' content='text/html; charset=UTF-8'>
        <style>
            iframe {
                display: none;
            }
        </style>
        </head>
        <body>
        <script type="text/javascript">
            function pageloaded(myframe) {
                document.getElementById("loading").style.display = "none";
                myframe.style.display = "inline";
        }
        </script>
        <p id='loading'>You start skimming through the manual...</p>
        <iframe width='100%' height='97%' onload="pageloaded(this)" src="http://tgstation13.org/wiki/Space_Law?printable=yes&remove_links=1" frameborder="0" id="main_frame"></iframe>
        </body>

        </html>
```
The Space law book, for example. Same results gained from other books that display text from the wiki that I tested in the dome.

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Fixes #79545

Basically, the TGUI markdown reader that every other book uses didn't properly send the HTML printified wiki page that we had. In order to fix this, I shuffled around a bunch of code and worked in an edge case to ensure that these manuals keep functioning. 

In an ideal world we would be able to do something more elegant, or maybe not even rely on displaying wiki pages in-game at all, but we just gotta play the hand we're dealt and ensure that these sorts of features continue to work for right now.

Also, we no longer need to both write and access raw HTML in the book framework to make everything work here. This should reduce a LOT of potential exploit vectors.

## Why It's Good For The Game

image

Thing work 👍 
## Changelog
:cl:
fix: Nanotrasen has clarified an issue with their manual publishers, and these guides should now contain actual user-pertinent content.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
why are there two scrollbars

File: code/game/objects/items/manuals.dm [hunk 1]
```diff
@@ -235,15 +235,20 @@
...
 
-/obj/item/book/manual/wiki/attack_self()
-	if(!book_data.content)
+/obj/item/book/manual/wiki/display_content(mob/living/user)
+	if(isnull(book_data.content))
 		initialize_wikibook()
-	return ..()
+
+	credit_book_to_reader(user)
+	var/book_content = book_data.get_content(trusted = TRUE) // legitimately one of the only times you should be using this argument
```
Comment (optimumtact) [inline code/game/objects/items/manuals.dm:250]: dont make it an argument make it another proc it's too dangerous to be left alive

Comment (optimumtact) [reply code/game/objects/items/manuals.dm:250]: and add a lint for existing only being called once

**Comment by san7890** [comment]
> why are there two scrollbars

One of them is for the Title "Space Law". Its stupid but no clue how to fix

File: code/game/objects/items/manuals.dm (hunk 1 above)
Comment (san7890) [reply code/game/objects/items/manuals.dm:250]: Maybe I should just access the content variable directly for this proc

Comment (san7890) [reply code/game/objects/items/manuals.dm]: even better i just made it a macro

Comment (optimumtact) [reply code/game/objects/items/manuals.dm]: yeah this is good.

**Review by optimumtact — APPROVED** [review]

**Pull request merged**